vnext/Desktop.IntegrationTests/WebSocketServer.cpp

```diff
@@ -0,0 +1,390 @@
+#include "WebSocketServer.h"
+
+#include <boost/asio/bind_executor.hpp>
+#include <boost/beast/core/buffers_to_string.hpp>
+
+using namespace boost::asio;
+
+using boost::system::error_code;
+using std::function;
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::string;
+
+namespace websocket = boost::beast::websocket;
+
+namespace Microsoft {
+namespace React {
+namespace Test {
+
+#pragma region BaseWebSocketSession
+
+template<typename SocketLayer>
+BaseWebSocketSession<SocketLayer>::BaseWebSocketSession(WebSocketServiceCallbacks& callbacks)
+  : m_callbacks{ callbacks }
+  , m_state{ State::Stopped }
+{
+}
+
+template<typename SocketLayer>
+BaseWebSocketSession<SocketLayer>::~BaseWebSocketSession()
+{
+}
+
+template<typename SocketLayer>
+void BaseWebSocketSession<SocketLayer>::Start()
+{
+  Accept();
+}
+
+template<typename SocketLayer>
+void BaseWebSocketSession<SocketLayer>::Accept()
+{
+  m_stream->async_accept_ex(
+    bind_executor(*m_strand, std::bind(
+      &BaseWebSocketSession<SocketLayer>::OnHandshake,
+      this->SharedFromThis(),
+      _1 // response
+    )),
+    bind_executor(*m_strand, std::bind(
+      &BaseWebSocketSession<SocketLayer>::OnAccept,
+      this->SharedFromThis(),
+      _1 // ec
+    ))
+  );
+}
+
+template<typename SocketLayer>
+void BaseWebSocketSession<SocketLayer>::OnHandshake(websocket::response_type& response)
+{
+  if (m_callbacks.OnHandshake)
+    m_callbacks.OnHandshake(response);
+}
+
+template<typename SocketLayer>
+void BaseWebSocketSession<SocketLayer>::OnAccept(error_code ec)
+{
+  if (ec)
+    return;//TODO: fail
```

Comment:
Is there an issue tracking the TODO?

Reply:
Same question applies to the other error-code-related TODOs in the file.

Reply:
Not yet. I have not figured out error handling, but this does not prevent basic functionality.